src/main/java/com/example/onlinebookstore/validation/category/impl/CategoryValidatorImpl.java

```diff
@@ -0,0 +1,37 @@
+package com.example.onlinebookstore.validation.category.impl;
+
+import com.example.onlinebookstore.dto.category.CategoryDto;
+import com.example.onlinebookstore.exception.InvalidCategoryException;
+import com.example.onlinebookstore.repository.category.CategoryRepository;
+import com.example.onlinebookstore.validation.category.CategoryValidator;
+import com.example.onlinebookstore.validation.common.NullFieldValidator;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class CategoryValidatorImpl implements CategoryValidator {
+    private final NullFieldValidator nullFieldValidator;
+    private final CategoryRepository categoryRepository;
+
+    @Override
+    public void isCategoryValid(CategoryDto categoryDto) {
+        nullFieldValidator.isFieldNotNull(categoryDto,
+                new InvalidCategoryException("Enter Category"));
+        isNameValid(categoryDto.getName());
+    }
+
+    @Override
+    public void isCreateCategoryValid(CategoryDto categoryDto) {
+        isCategoryValid(categoryDto);
+        if (categoryRepository.findByName(categoryDto.getName()).isPresent()) {
+            throw new InvalidCategoryException("Category with such name is already exists");
```

Comment:
1) I'd make this more generic. Like EntityNotFoundException.
would be something like EntityAlreadyPresentException
most probably it's not the last case when you need to handle possible conflict
2) Where is GlobalExceptionHandler logic for newly introduced unchecked exception?

Reply:
thank you added new exception